### PR TITLE
[story/ECP-908] - change to using correct award amounts in eligible later 2022 screen

### PR DESCRIPTION
**Pull request by WorkSutton**
Show the base amount and the uplifted amount based on subject and academic ITT year
Add details of criteria in an initializer file

### CODE REVIEW

**Review by martikat — APPROVED** [review]

File: app/helpers/claims/show_helper.rb [hunk 1]
```diff
@@ -11,5 +11,21 @@ def claim_submitted_title(claim)
     def shared_view_css_class_size(claim)
       claim.has_ecp_policy? ? "l" : "xl"
     end
+
+    def base_and_uplift_award_amounts(claim, year)
```
Comment (nomasprime) [inline app/helpers/claims/show_helper.rb:15]: Seems like this should be named after an object that it returns, ie, missing object.

File: app/helpers/claims/show_helper.rb [hunk 2]
```diff
@@ -11,5 +11,21 @@ def claim_submitted_title(claim)
...
+      return nil unless claim.has_ecp_policy?
+
+      base_amount = MATRIX.dig(claim.eligibility.eligible_itt_subject, claim.eligibility.itt_academic_year, year)
```
Comment (nomasprime) [inline app/helpers/claims/show_helper.rb:18]: Not obvious what this does, missing method?

File: app/models/early_career_payments/eligibility_matrix_calculator.rb [hunk 3]
```diff
@@ -1,12 +1,5 @@
 module EarlyCareerPayments
   class EligibilityMatrixCalculator
```
Comment (nomasprime) [inline app/models/early_career_payments/eligibility_matrix_calculator.rb:2]: Didn't mention on previous PR but I think in future we could do better on consolidating and naming eligibility related code.

File: app/models/early_career_payments/eligibility_matrix_calculator.rb [hunk 4]
```diff
@@ -16,7 +9,7 @@ def initialize(eligibility)
     def eligible_later?
       return false if subject == :mathematics && itt_academic_year == "2018_2019"
```
Comment (nomasprime) [inline app/models/early_career_payments/eligibility_matrix_calculator.rb:10]: Also, didn't mention feels odd having mixture of symbols and strings here. These could be missing objects.

File: app/models/early_career_payments/eligibility_matrix_calculator.rb [hunk 5]
```diff
@@ -16,7 +9,7 @@ def initialize(eligibility)
...
 
-      MATRIX.dig(subject)&.include?(itt_academic_year) ? true : false
+      MATRIX.dig(subject.to_s, itt_academic_year).nil? ? false : true
```
Comment (nomasprime) [inline app/models/early_career_payments/eligibility_matrix_calculator.rb:12]: Didn't say anything before given current situation but glad you changed this 👍

File: config/initializers/early_career_payments/eligibility_matrix_calculator.rb [hunk 6]
```diff
@@ -0,0 +1 @@
+MATRIX = YAML.safe_load(File.read(File.expand_path("../eligibility_matrix.yml", __FILE__))).freeze
```
Comment (nomasprime) [inline config/initializers/early_career_payments/eligibility_matrix_calculator.rb:1]: Why YAML not inline Ruby?

File: spec/helpers/claims/show_helper_spec.rb [hunk 7]
```diff
@@ -38,4 +38,76 @@
...
+
+      it "returns nil" do
+        expect(helper.base_and_uplift_award_amounts(claim, nil)).to be_nil
+      end
+    end
+
+    context "with a EarlyCareerPayments policy" do
+      let(:policy) { EarlyCareerPayments }
+
+      describe "Chemistry/Physics/Foreign Languages" do
```
Comment (nomasprime) [inline spec/helpers/claims/show_helper_spec.rb:54]: Again, this kind of thing should be context.

**Review by nomasprime — APPROVED** [review]
Few things but, again, given current direction I'll approve.

Quite a few things we could improve on the specs but didn't comment because we'll probably have to revisit later anyway.

File: app/models/early_career_payments/eligibility_matrix_calculator.rb (hunk 3 above)
Comment (WorkSutton) [reply app/models/early_career_payments/eligibility_matrix_calculator.rb:2]: Yes I agree, eligibility is everywhere and its been a challenge as to what to call some methods/classes. I went with the MatrixCalculator as it calcuates if a ECP claims eligibility is actually one of 3 statuses after all of the eligibility questions have been successfully answered. These are:

- Eligible Now (in this claim window/AcademicYear)
- Eligible Later (say in 2022 if in 2021 and meets all criteria)
- No longer eligible

File: app/models/early_career_payments/eligibility_matrix_calculator.rb (hunk 5 above)
Comment (WorkSutton) [reply app/models/early_career_payments/eligibility_matrix_calculator.rb:12]: @nomasprime Yes I changed it as I changed the data structure to better reflect additional information. Its definitely better just using the `dig` method.

File: spec/helpers/claims/show_helper_spec.rb (hunk 7 above)
Comment (WorkSutton) [reply spec/helpers/claims/show_helper_spec.rb:54]: Is it ok to have context, context, context? I thought the linters got fussy and liked 2 levels of describes, then multiple contexts if more than 3 levels deep.
Maybe thats rubocop which we are overriding with `standard` which I have to say personally I am not a fan of

**Pull request merged**